### PR TITLE
c_sharp_exports.rst: clarify why exporting nodes/resources is useful

### PR DESCRIPTION
**Pull request by 31**
It's easy to miss how useful this is! It's is in my unwritten top 10 of 4.x features. 🙂

### CODE REVIEW

File: tutorials/scripting/c_sharp/c_sharp_exports.rst [hunk 1]
```diff
@@ -81,6 +83,17 @@ Resources and nodes can be exported.
     [Export]
     public Node Node { get; set; }
 
+Exporting a specific type of resource or node lets the property editor show a
+filtered list of possibilities.
```
[the remark this answers is not in the document]
Comment (31) [reply tutorials/scripting/c_sharp/c_sharp_exports.rst]: I think the Node and Resource sections are pretty bare themselves. I'm able to infer that if you're able to export a Node, then anything that "is" a Node should be able to be exported too, but even for me, I second-guess whether that's what the doc actually means or if that inference will bite me.

To get peoples' eyes on it while keeping the intro simple, how about removing `[Export] public Resource Resource { get; set; }` in the intro and go straight to `[Export] public PackedScene PackedScene { get; set; }` with a link to the section for more info? Pushed an attempt at that.

[the remark this answers is not in the document]
Comment (raulsntos) [reply tutorials/scripting/c_sharp/c_sharp_exports.rst]: That works for me :+1:

**Review by raulsntos — APPROVED** [review]
LGTM. Thanks!

**Review by skyace65 — APPROVED** [review]

**Comment by skyace65** [comment]
Thanks!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.2 in https://github.com/godotengine/godot-docs/pull/9648.

**Pull request merged**